docs/webapp.adoc

```diff
@@ -1,30 +1,625 @@
+        </style>
+    </head>
+
+    <body>
+        <h1>Top 3 movies to put in a list</h1>
+        <p>#4 will blow your mind!</p>
+
+        <!--4-->
+        <div id="movieListContainer">
+            <div class="faux-spinner">Loading movies...</div>
+        </div>
+
+        <!--5-->
+        <script>
+            var MOVIE_LIST_PARAMS= {
+                parentPath: '/moviesite',
+                apiUrl: '/admin/site/preview/default/draft/moviesite/api/headless',
+                movieType: 'com.enonic.app.react4xp:movie',
+                movieCount: 3,
+                sortExpression: 'data.year ASC',
+            };
+        </script>
+
+        <!--6-->
+        <script defer src="(...) /webapp/script.js"></script>
+    </body>
+</html>
+
+----
+<1> We start by running React and ReactDOM from a CDN.
+<2> Next, we fetch 3 dependency chunks that the _MovieList_ entry needs: _shared.&lt;hash&gt;.js_, _shared.&lt;hash&gt;.css_ and _MovieList.css_.
++
+[NOTE]
+====
+As before, these are asset URLs that depend on your setup - and this applies to both the *content-dependent &lt;hash&gt;* in the filename and the *asset path* itself. On my local machine running XP, the path (the `(...)` before _/react4xp/..._) is this:
+
+`/webapp/com.enonic.app.react4xp/_/asset/com.enonic.app.react4xp:1604314030`
+
+If you've run through the previous chapter, you can for example copy the corresponding asset-URLs from the produced page source HTML in the preview.
+====
+<3> The _MovieList_ entry asset. Loading and running this will expose the entry in the global JS namescape as a function that creates a react component: `React4xp['MovieList'].default`.
+<4> The target container for the react app. The not-really-a-spinner (_"Loading movies..."_) will be replaced when the actual _MovieList_ is rendered into the container.
+<5> Variables used by our particular script later, just wrapped in a `MOVIE_LIST_PARAMS` object to encapsulate them in the global namespace. These are the same values as in in the previous chapter, and the script at the end will use these in a `props` object, to create the renderable react app from the _MovieList_ entry.
++
+Also note that we just hardcoded the values of `parentPath`, `apiUrl` and `movieType` here - they may be different in your setup. As <<guillotine#, before>>: `parentPath` is the content path of the site item under which the movie items will be found, `apiUrl` is the full path to `/api/headless` _below that site_ (recall that the URL to the guillotine API depends on the URL of a site item like this, because of the way we set the API up with a controller mapping <<guillotine#expose_api, earlier>>), and `movietype` is the full, appname-dependent content type of the movie items to look for.
+<6> Finally, loading the compiled script from _script.es6_ <<#script, below>>.
+
 {zwsp} +
 
+A *webapp controller* is needed for XP to serve this HTML, and it's about as minimal as an XP controller can be:
 
-== Blabla
+.webapp.es6:
```

Comment:
The shortest xp controller ever created 😆

Reply:
🔬